modules/programs/gh-dash.nix

```diff
@@ -0,0 +1,40 @@
+{ config, lib, pkgs, ... }:
+
+let
+
+  cfg = config.programs.gh-dash;
+
+  yamlFormat = pkgs.formats.yaml { };
+
+in {
+  meta.maintainers = [ lib.maintainers.janik ];
+
+  options.programs.gh-dash = {
+    enable = lib.mkEnableOption "GitHub CLI dashboard plugin";
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.gh;
+      defaultText = lib.literalExpression "pkgs.gh";
+      description = "Package providing <command>gh</command>.";
+    };
+
+    settings = lib.mkOption {
+      type = yamlFormat.type;
+      default = { };
+      description =
+        "Configuration written to <filename>$XDG_CONFIG_HOME/gh-dash/config.yml</filename>.";
+      example = lib.literalExpression ''
+        {
+        };
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
```

Comment:
I think this should also put the extension manifest in
```
$HOME/.local/share/gh/extensions/gh-dash/manifest.yml
```
having the content:
```
path: ${cfg.package}/bin/gh-dash
```
so that `gh dash` will work in addition to `gh-dash`.

Reply:
I'm currently using the module with `gh dash`. Just tested `gh dash` and `gh-dash` both work fine.

Reply:
Hmm... do you have anything in `$HOME/.local/share/gh/extensions/gh-dash/manifest.yml`?

Reply:
nope
```
tree ~/.local/share/gh/extensions/               
 /home/janik/.local/share/gh/extensions
└──  gh-dash -> /nix/store/ga0gj4lrgis1xib4yajn4q9gx8qxcvd9-gh-dash-3.7.9/bin
```

Reply:
I also have this in my hm config maybe that matters :shrug: 
```nix
{
  programs.gh = {
    enable = true;
    extensions = with pkgs; [
      gh-dash
    ];
  };
}
```

I should probably add:
```nix
{
  home.packages = [ pkgs.gh-dash ];
  programs.gh.extensions = [ pkgs.gh-dash ];
}
```
to the module enable option?

Reply:
If this currently blocking the PR? If so I  can add the `$HOME/.local/share/gh/extensions/gh-dash/manifest.yml` path to the config if it makes you happy.